vaultlocker/shell.py

```diff
@@ -69,19 +70,50 @@ def _encrypt_block_device(args, client, config):
     block_uuid = str(uuid.uuid4()) if not args.uuid else args.uuid
     vault_path = _get_vault_path(block_uuid, config)
 
-    dmcrypt.luks_format(key, block_device, block_uuid)
-    # Ensure sym link for new encrypted device is created
-    # LP Bug #1780332
-    dmcrypt.udevadm_rescan(block_device)
-    dmcrypt.udevadm_settle(block_uuid)
-
-    # NOTE: store and validate key
-    client.write(vault_path,
-                 dmcrypt_key=key)
-    stored_data = client.read(vault_path)
-    assert key == stored_data['data']['dmcrypt_key']
+    # NOTE: store and validate key before trying to encrypt disk
+    try:
+        client.write(vault_path,
+                     dmcrypt_key=key)
+    except hvac.exceptions.VaultError as write_error:
+        logger.error(
+            'Vault write to path {}. Failed with error:{}'.format(
+                vault_path, write_error))
+        raise exceptions.VaultWriteError(vault_path, write_error)
 
-    dmcrypt.luks_open(key, block_uuid)
+    try:
+        stored_data = client.read(vault_path)
+    except hvac.exceptions.VaultError as read_error:
+        logger.error('Vault access to path {}'
+                     'failed with error:{}'.format(vault_path, read_error))
+        raise exceptions.VaultReadError(vault_path, read_error)
+
+    if not key == stored_data['data']['dmcrypt_key']:
+        raise exceptions.VaultKeyMismatch(vault_path)
```

Comment:
Unit test coverage for the above line is OK (switch from assertion to VaultKeyMismatch)